seqr/views/apis/project_api_tests.py

```diff
@@ -166,3 +170,62 @@ def test_empty_project_page_data(self):
         self.assertDictEqual(response_json['locusListsByGuid'], {})
+
+
+# Tests for AnVIL access disabled
+class LocalProjectAPITest(AuthenticationTestCase, TestHelpers.ProjectAPITest):
+    fixtures = ['users', '1kg_project', 'reference_data']
+    PROJECT_COLLABORATORS = [{'dateJoined': '2017-03-12T23:09:54.180Z', 'displayName': 'Test Manager User', 'email': 'test_user_manager@test.com', 'firstName': 'Test Manager User', 'hasEditPermissions': True, 'hasViewPermissions': True, 'id': 11, 'isAnvil': False, 'isStaff': False, 'lastName': '', 'username': 'test_user_manager'},
+{'dateJoined': '2017-03-12T23:09:54.180Z', 'displayName': 'Test Non Staff User', 'email': 'test_user_no_staff@test.com', 'firstName': 'Test Non Staff User', 'hasEditPermissions': False, 'hasViewPermissions': True, 'id': 12, 'isAnvil': False, 'isStaff': False, 'lastName': '', 'username': 'test_user_non_staff'}]
+
+
+ANVIL_PROJECT_COLLABORATORS = [{'dateJoined': '2017-03-12T23:09:54.180Z', 'displayName': 'Test Manager User', 'email': 'test_user_manager@test.com', 'firstName': 'Test Manager User', 'hasEditPermissions': False, 'hasViewPermissions': True, 'id': 11, 'isAnvil': True, 'isStaff': False, 'lastName': '', 'username': 'test_user_manager'},
+{'dateJoined': '2017-03-12T23:09:54.180Z', 'displayName': 'Test Non Staff User', 'email': 'test_user_no_staff@test.com', 'firstName': 'Test Non Staff User', 'hasEditPermissions': False, 'hasViewPermissions': True, 'id': 12, 'isAnvil': True, 'isStaff': False, 'lastName': '', 'username': 'test_user_non_staff'}]
+
+
+# Test for permissions from AnVIL only
+class AnvilProjectAPITest(AnvilAuthenticationTestCase, TestHelpers.ProjectAPITest):
+    fixtures = ['users', 'social_auth', '1kg_project', 'reference_data']
+    PROJECT_COLLABORATORS = ANVIL_PROJECT_COLLABORATORS
+
+    def test_create_update_and_delete_project(self):
+        super(AnvilProjectAPITest, self).test_create_update_and_delete_project()
+        self.mock_list_workspaces.assert_not_called()
+        self.mock_service_account.get.assert_not_called()
+
+    def test_project_page_data(self):
+        super(AnvilProjectAPITest, self).test_project_page_data()
+        self.mock_list_workspaces.assert_not_called()
+        self.mock_service_account.get.assert_called_with(
+            'api/workspaces/my-seqr-billing/anvil-1kg project n\u00e5me with uni\u00e7\u00f8de/acl')
+        self.assertEqual(self.mock_service_account.get.call_count, 5)
+
+    def test_empty_project_page_data(self):
+        url = reverse(project_page_data, args=[EMPTY_PROJECT_GUID])
+        self.check_collaborator_login(url)
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 403)
+
+
+# Test for permissions from AnVIL and local
+class MixProjectAPITest(MixAuthenticationTestCase, TestHelpers.ProjectAPITest):
+    fixtures = ['users', 'social_auth', '1kg_project', 'reference_data']
+    PROJECT_COLLABORATORS = ANVIL_PROJECT_COLLABORATORS
```

Comment:
For the mix case, is there any way to test that it combines permissions? Its okay not to if thats really hard

Reply:
It should be possible.

Reply:
I add a pure local user to make the mix-case has a different collaborator list.